super-pane/search-field/index.tsx

```diff
@@ -0,0 +1,47 @@
+import { TextInput } from '@sanity/ui'
+import React, { useEffect, useState } from 'react'
+import FieldToSearchFor from './FieldToSearchFor';
+import SelectFieldToSearchFor from './select-field-to-search-for'
+import styles from './styles.module.css';
+
+interface SearchFieldProps {
+  searchCallback: (userQuery: string) => void
+  fieldsToChooseFrom: FieldToSearchFor[],
+  currentField: FieldToSearchFor,
+  onFieldSelected: (field: FieldToSearchFor) => void
+}
+
+let timeout: NodeJS.Timeout
```

Comment:
In my experience, variance in types between browser and Node can sometimes cause issues so I usually assert it as `ReturnType<typeof setTimeout>`

```suggestion
let timeout: ReturnType<typeof setTimeout> | undefined
```

Reply:
Oh, thats a valid point. Didn't think of that. Clever 🤠